libs/wingsdk/src/util/util.ts

```diff
@@ -259,9 +270,19 @@ export class Util {
   /**
    * Suspends execution for a given duration.
    * @param delay The time to suspend execution.
-   * @inflight
+   * @inflightImpl __sleep__inflight
    */
-  public static async sleep(delay: Duration): Promise<void> {
+  public static sleep(delay: Duration): void {
+    Atomics.wait(
+      new Int32Array(new SharedArrayBuffer(4)),
+      0,
+      0,
+      delay.seconds * 1000
+    );
```

Comment:
When I first saw the Atomics API I got excited because I assumed it was a great hack for synced-async, and then came the sadness of the big limitation.

It's cool that this works now even though I think we'll eventually get off of worker threads and have to use a stinky execSync for sleep.